### DISCUSSION TITLE
saving retained messages

### DISCUSSION THREAD

**Discussion by Do4bled** [dotnet/MQTTnet#1438] [Q&A] [answered]
Is there an example/tutorial or somebody who implemented something to save retained messages with the latest version? I want to store all the retained messages so that when the Server restarts all the messages can be put back. 

I've seen some examples but they seem outdated or not relevant any longer. Any help will be appreciated.

**Comment by chkr1011** [comment, marked as answer]
I just created the sample "Persist_Retained_Messages". Please let me know if it works for you.

**Reply by Do4bled** [reply]
Thanks, this workes well.
